src/rpc/src/chainStorageWatcher.ts

```diff
@@ -0,0 +1,167 @@
+import { AgoricChainStoragePathKind } from './types';
+import { batchVstorageQuery } from './batchQuery';
+import type { UpdateHandler } from './types';
+import type { Unserialize } from '@endo/marshal';
+
+type Subscriber<T> = {
+  onUpdate: UpdateHandler<T>;
+};
+
+const defaultNewPathQueryDelayMs = 20;
+const defaultRefreshLowerBoundMs = 2000;
+const defaultRefreshUpperBoundMs = 4000;
+
+const randomRefreshPeriod = (
+  refreshLowerBoundMs: number,
+  refreshUpperBoundMs: number,
+) =>
+  Math.floor(Math.random() * (refreshUpperBoundMs - refreshLowerBoundMs)) +
```

Comment:
floor seems oddly specific. wouldn't `round` work?

also curious: does something fail if this returns a floating point?